base/shared/src/main/scala/scalaz/tc/foldable.scala

```diff
@@ -6,13 +6,24 @@ import scala.List
 import scala.language.experimental.macros
 
 trait FoldableClass[F[_]] {
+  /**
+  Applies a binary operator to a start value and all elements of this sequence, going left to right.
+  Note: this will not terminate for infinite-sized collections. For Example, Stream.
+  */
```

Comment:
Great, thank you. Making note of the evaluation semantic would be nice (i.e. evaluates strictly, lazily, etc) instead of saying "this will not terminate". For example, what if my Stream is finite? It works just fine! It's enough to say it is lazy/strict.